File: src/shared/kartEngine/KartMove/FirstPipelineValue.ts
/**
 * FirstPipelineValue struct的TypeScript实现
 * 与Lua版本完全一致，用于处理第一管道值
 */

import { UnityVector3 } from "../KartShared/UnityMath";

export class FirstPipelineValue {
    public left: UnityVector3 = new UnityVector3(0, 0, 0);
    public up: UnityVector3 = new UnityVector3(0, 0, 0);
    public front: UnityVector3 = new UnityVector3(0, 0, 0);
    public frontVel: number = 0.0;
    public leftVel: number = 0.0;
    public upVel: number = 0.0;
    public speed: number = 0.0;

    constructor() {
        this.Initialize();
    }

    public Initialize(): void {
        this.left = new UnityVector3(0, 0, 0);
        this.up = new UnityVector3(0, 0, 0);
        this.front = new UnityVector3(0, 0, 0);
        this.frontVel = 0.0;
        this.leftVel = 0.0;
        this.upVel = 0.0;
        this.speed = 0.0;
    }

    // 静态工厂方法
    static create(): FirstPipelineValue {
        return new FirstPipelineValue();
    }
}

export default FirstPipelineValue;

File: sources/roblox-lua-items-network/shared/kartEngine/KartMove/FirstPipelineValue.lua
-- FirstPipelineValue struct的Lua等效实现
local UnityMath = require(script.Parent.Parent.KartShared.UnityMath)
local Vector3 = UnityMath.Vector3

local FirstPipelineValue = {}
FirstPipelineValue.__index = FirstPipelineValue

function FirstPipelineValue.new()
    local self = setmetatable({}, FirstPipelineValue)
    
    self.left = Vector3.new()
    self.up = Vector3.new()
    self.front = Vector3.new()
    self.frontVel = 0.0
    self.leftVel = 0.0
    self.upVel = 0.0
    self.speed = 0.0
    
    return self
end

return FirstPipelineValue